src/emailcontactmodel.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2020 - 2021 Open Mobile Platform LLC.
+ *
+ * This program is licensed under the terms and conditions of the
+ * Apache License, version 2.0.  The full text of the Apache License is at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#ifndef EMAILCONTACTLISTMODEL_H
+#define EMAILCONTACTLISTMODEL_H
+
+#include <QAbstractListModel>
+#include <QHash>
+#include <QList>
+#include <QPair>
+#include <QMailAddress>
+#include <QMailTimeStamp>
+#include <qmailstore.h>
+
+
+class Q_DECL_EXPORT EmailContactModel : public QAbstractListModel
+{
+    Q_OBJECT
+
+public:
+    enum Role {
+        NameRole = Qt::UserRole + 1,
+        EmailRole,
+        TimeStampRole,
+    };
+
+    explicit EmailContactModel(QAbstractItemModel *parent = nullptr);
+    ~EmailContactModel() override = default;
+
+    int rowCount(const QModelIndex &parent = QModelIndex()) const override;
+    QVariant data(const QModelIndex &index, int role = Qt::DisplayRole) const override;
+
+    Q_INVOKABLE void initialize(const int maxCount);
```

Comment:
Such a thing still shouldn't exist on the api and not spotting any usage either.